hydra-server/src/main/java/com/pandora/hydra/server/partition/PartitionService.java

```diff
@@ -166,6 +173,13 @@ private TestContainer collapsePartition(Partition partition) {
                 .collect(Collectors.toSet());
     }
 
+    private Set<String> buildTestBlacklist(TestRun testRun, String forHost, String forProject) {
+        return testRun.getPartitions().stream()
```

Comment:
I hadn't until just now. I'm a little hesitant to arbitrarily parallelize something whose inner workings I'm not entirely familiar with, for fear of introducing subtle concurrency problems. I just followed @jguerra 's example from his existing `buildTestBlacklist()` method, which also used the sequential `stream()`. I'm not too familiar with Java 8's streaming operations, so what I've read seems to suggest that there is some additional overhead for establishing/synchronizing the parallel streams, and unless we're encountering a performance problem due to a _massive_ number of items and/or significant time to process each item, it may be better to stick with sequential. To my knowledge, generation of the blacklists doesn't appear to be a bottleneck - maybe we revisit this when we get to that point?

Reply:
The actual limitation is that underlying thread pool is only 50 threads. So if you are using parallelStream too hard - it might become a bottleneck (thus require more advanced multi threading techniques). Though in our case we don't have that many requests and probably can benefit from parallel stream. @jguerra Thoughts?

Reply:
Justin is on vacation all week :) I've opened a separate ticket to try out `parallelStream()`, as I feel a change of that nature is out of scope for this particular PR, and would require extensive testing.